src/v/archival/ntp_archiver_service.cc

```diff
@@ -382,9 +410,29 @@ ntp_archiver::upload_segment(upload_candidate candidate) {
       _segment_upload_timeout, _cloud_storage_initial_backoff, &_rtcnode);
     retry_chain_logger ctxlog(archival_log, fib, _ntp.path());
 
-    auto path = cloud_storage::generate_remote_segment_path(
-      _ntp, _rev, candidate.exposed_name, _start_term);
+    auto key = cloud_storage::parse_segment_name(candidate.exposed_name);
```

Comment:
Good point, it looks like just storing the segment term in the candidate might be enough to avoid the parse as that is the only information missing from the struct.

The tricky code path is when we are concatenating segments, but storing the term for the first segment which is stored in `candidate.sources` looks like the consistent thing to do there (that is what happens after the parse right now).

Reply:
added the segment term to upload candidate